### PR TITLE
Test/fixing unit tests

### LINKED ISSUE #239: fix or delete failing unit tests in bbi repo
* once #237 finishes we should at least have a clean list of existing tests and their status. (Comment [here](https://github.com/metrumresearchgroup/bbi/issues/237#issuecomment-921044628))
* hopefully #238 will give us some rough guidelines for whether certain tests are necessary. In other words, some criteria to say "this test is failing and we don't strictly need it so let's just delete it."
* ~~add [`mrgvalidate`-style Test ID's](https://metrumresearchgroup.github.io/mrgvalidate/reference/input_formats.html) to tests as we go.~~ Just going to do this all in https://github.com/metrumresearchgroup/bbi/issues/246
* output of this issue should be a passing drone build and passing `go list ./... | grep -v integration | xargs go test` locally (because integration tests won't run locally).

### PR DESCRIPTION
**Pull request by seth127**
Closes #239

### CODE REVIEW

**Comment by seth127** [comment]
@elisarver feel free to take a look, and to start on #244 as soon as this is merged. But I do think we should have @dpastoor give this a lookover to make sure I didn't remove anything that he thinks should stay.

**Comment by seth127** [comment]
@dpastoor I tried to keep the commits pretty clean, in terms of which tests/files they address, so that might be the easier way to go through this, as opposed to looking at the raw file diff view.

File: parsers/nmparser/set_missing_values_to_default_test.go [hunk 1]
```diff
@@ -128,22 +116,21 @@ func TestSetMissingValuesToDefaultParameterDataValues(tt *testing.T) {
...
 		tt.Run(test.name, func(tt *testing.T) {
 			t := wrapt.WrapT(tt)
 
-			setMissingValuesToDefault(&test.modelOutput, test.etaCount, test.epsCount)
-			t.R.Equal(7, len(test.modelOutput.ParametersData[0].StdErr.Theta), "Fail :"+test.context)
-			t.R.Equal(2, len(test.modelOutput.ParametersData[0].RandomEffectSD.Sigma), "Fail :"+test.context)
-			t.R.Equal(3, len(test.modelOutput.ParametersData[0].RandomEffectSDSE.Omega), "Fail :"+test.context)
+			setMissingValuesToDefault(&test.modelOutput)
+			t.R.Equal(7, len(test.modelOutput.ParametersData[0].StdErr.Theta), "Fail :"+test.name)
+			t.R.Equal(2, len(test.modelOutput.ParametersData[0].RandomEffectSD.Sigma), "Fail :"+test.name)
```
Comment (elisarver) [inline parsers/nmparser/set_missing_values_to_default_test.go:128]: test.name is covered in all of these cases by the tt.Run above. it is reported as `Test<Name>/7_thetas_3_omegas_2_sigmas`

**Review by elisarver — APPROVED** [review]
Happy to help with the clean-up ahead of this, good changes.

File: parsers/nmparser/set_missing_values_to_default_test.go (hunk 1 above)
Comment (seth127) [reply parsers/nmparser/set_missing_values_to_default_test.go:128]: ah, so can we just get rid of that `, "Fail :"+test.name` part of each line?

Comment (elisarver) [reply parsers/nmparser/set_missing_values_to_default_test.go:128]: yup!

**Pull request merged**